server/cluster/cluster.go

```diff
@@ -16,15 +19,17 @@ import (
 
 // Server provides a Cluster service
 type Server struct {
-	db  db.ArgoDB
-	enf *rbac.Enforcer
+	kubeclientset kubernetes.Interface
+	db            db.ArgoDB
+	enf           *rbac.Enforcer
 }
 
 // NewServer returns a new instance of the Cluster service
-func NewServer(db db.ArgoDB, enf *rbac.Enforcer) *Server {
+func NewServer(kubeclientset kubernetes.Interface, db db.ArgoDB, enf *rbac.Enforcer) *Server {
```

Comment:
Per comment below, cluster server should not need a kubeclientset.  We create a clientset on the fly from a kubeconfig.